### DISCUSSION TITLE
zsh - was working, not working now

### DISCUSSION THREAD

**Discussion by macxcool** [starship/starship#1985] [Q&A] [answered]
I put the proper line in my .zshrc.local file and it was working. Now it isn't. Starship works in bash and pwsh and fish, but not in zsh. The only config I've done is a custom module, but it works in the other shells just fine. What should I check?

**Comment by mickimnet** [comment, marked as answer]
What does your prompt in zsh show? And, is `eval "$(starship init zsh)"` still in your `.zshrc`?

**Reply by macxcool** [reply]
It's just my regular zsh prompt without any starship elements.

**Reply by mickimnet** [reply]
How does your `.zshrc` look like? Is `eval "$(starship init zsh)"` still in it?

**Reply by macxcool** [reply]
Yup. It's there. I first put it there. Then I 'source'ed it an starship was working. Now I can't get anything but my regular prompt in zsh.  Fish, bash and pwsh all get the starship prompt just fine.
I'm using a more complex zsh setup which sources. .zsh.local and that's where I put my own customizations. It works fine for everything else I have in there, though.

**Reply by mickimnet** [reply]
What happens if you manually run `eval "$(starship init zsh)"`?

**Reply by macxcool** [reply]
No error, but nothing changes.
and 'starship explain' works fine too. It tells me what should be there, even in different contexts.

**Reply by macxcool** [reply]
I'm using the grml zsh config. It has a .zshrc.local and .zshrc.pre for your own customizations. I tried putting the starship line in .zshrc.pre too, but that didn't help.

**Reply by macxcool** [reply]
I'm going to try just using my config without the grml stuff.

**Reply by macxcool** [reply]
I took my .zshrc.local and made it my .zshrc, bypassing the grml config. That works. I get the starship prompt as I've configured it.
That sucks. I like that config setup. I wonder what's going on. And why did it work at first?

**Reply by macxcool** [reply]
Thanks @mickimnet for the push. I discovered that I had to updated my .zshrc to match changes in the grml zsh setup. This wasn't starship's fault at all (I didn't think it was, but I didn't know what had happened). Now everything works fine. I really like this, and it was so incredibly easy to get the same prompt in all shells.

**Reply by mickimnet** [reply]
@macxcool, you‘re welcome. Glad my hints helped you out. I believe that your grml setup set‘s the prompt somewhere as well. Probably after your _Starship_ entry.
